irods/test/data_obj_test.py

```diff
@@ -1028,6 +1028,61 @@ def test_register_with_xml_special_chars(self):
         # delete file
         os.remove(test_file)
 
+    def test_get_data_objects(self):
+        # Can't do one step open/create with older servers
+        if self.sess.server_version <= (4, 1, 4):
+            self.skipTest('For iRODS 4.1.5 and newer')
+
+        # test vars
+        test_dir = '/tmp'
+        filename = 'get_data_objects_test_file'
+        test_file = os.path.join(test_dir, filename)
+        collection = self.coll.path
+
+        # make random 16byte binary file
+        original_size = 16
+        with open(test_file, 'wb') as f:
+            f.write(os.urandom(original_size))
+
+        # make ufs resources
+        ufs_resources = []
+        for i in range(2):
+            resource_name = 'ufs{}'.format(i)
+            resource_type = 'unixfilesystem'
+            resource_host = self.sess.host
+            resource_path = '/tmp/{}'.format(resource_name)
+            ufs_resources.append(self.sess.resources.create(
+                resource_name, resource_type, resource_host, resource_path))
+
+        # put file in test collection and replicate
+        obj_path = '{collection}/{filename}'.format(**locals())
+        options = {kw.DEST_RESC_NAME_KW: ufs_resources[0].name}
+        self.sess.data_objects.put(test_file, collection + '/', **options)
```

Comment:
For some reason, I'm getting a KeyError when running the test as-is:
```
$ python -m unittest data_obj_test.TestDataObjOps.test_get_data_objects
E
======================================================================
ERROR: test_get_data_objects (data_obj_test.TestDataObjOps)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "data_obj_test.py", line 1061, in test_get_data_objects
    self.sess.data_objects.put(test_file, 'collection' + '/', **options)
  File "/var/lib/irods/python-irodsclient/irods/manager/data_object_manager.py", line 70, in put
    with open(file, 'rb') as f, self.open(obj, 'w', **options) as o:
  File "/var/lib/irods/python-irodsclient/irods/manager/data_object_manager.py", line 154, in open
    desc = conn.recv().int_info
  File "/var/lib/irods/python-irodsclient/irods/connection.py", line 95, in recv
    raise get_exception_by_code(msg.int_info)
  File "/var/lib/irods/python-irodsclient/irods/exception.py", line 70, in get_exception_by_code
    return iRODSExceptionMeta.codes[code](message)
KeyError: -130000

----------------------------------------------------------------------
Ran 1 test in 0.372s

FAILED (errors=1)
```
Changing the logical path to `'{collection}/'.format(**locals())` fixes it right up.